src/shared_modules/utils/socketDBWrapper.hpp

```diff
@@ -43,8 +45,49 @@ class SocketDBWrapper final
             [&](const char* body, uint32_t bodySize, const char* header, uint32_t headerSize)
             {
                 std::unique_lock<std::mutex> lock {m_mutex};
-                m_response = std::string(body, bodySize);
-                m_conditionVariable.notify_one();
+                std::string responsePacket(body, bodySize);
+
+                if (responsePacket.empty())
```

Comment:
Coding style: this is only a suggestion.
We could avoid writing `m_conditionVariable.notify_one()` every time is we simply change the order

<details><summary>Details</summary>
<p>

```c++
                if (0 == responsePacket.compare(0, 3, DB_WRAPPER_DUE)) {
                    m_responsePartial.push_back(nlohmann::json::parse(responsePacket.substr(4)));
                } else {
                    if (responsePacket.empty())
                    {
                        m_exceptionStr = "Empty DB response";
                    }
                    else if (0 == responsePacket.compare(0, 3, DB_WRAPPER_ERROR))
                    {
                        m_exceptionStr = "DB query error: " + responsePacket.substr(4);
                    }
                    else if (0 == responsePacket.compare(0, 3, DB_WRAPPER_IGNORE))
                    {
                        m_exceptionStr = "DB query ignored: " + responsePacket.substr(4);
                    }
                    else if (0 == responsePacket.compare(0, 3, DB_WRAPPER_UNKNOWN))
                    {
                        m_exceptionStr = "DB query unknown response: " + responsePacket.substr(4);
                    }
                    else if (0 == responsePacket.compare(0, 2, DB_WRAPPER_OK))
                    {
                        if(!m_responsePartial.empty())
                        {
                            m_response = m_responsePartial;
                        }
                        else
                        {
                            m_response = nlohmann::json::parse(responsePacket.substr(3));
                        }
                    }
                    else
                    {
                        m_exceptionStr = "DB query invalid response: " + responsePacket;
                    }
                    m_conditionVariable.notify_one();
                }
```



</p>
</details>